modules/cudf/src/table/concatenate.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021, NVIDIA CORPORATION.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cudf/concatenate.hpp>
+#include <node_cudf/table.hpp>
+
+namespace nv {
+
+Napi::Value Table::concat(Napi::CallbackInfo const& info) {
+  CallbackArgs args{info};
+
+  std::vector<Table::wrapper_t> tables = args[0];
+  rmm::mr::device_memory_resource* mr  = args[1];
+
+  std::vector<cudf::table_view> table_views;
+  table_views.reserve(tables.size());
+  std::transform(tables.begin(),
+                 tables.end(),
+                 std::back_inserter(table_views),
+                 [](const Table::wrapper_t& t) -> cudf::table_view { return t->view(); });
+
+  return Table::New(info.Env(), cudf::concatenate(table_views, mr))->Value();
```

Comment:
I think this needs a `try...catch` since `cudf::concatenate` can potentially throw a cudf::logic_error:

https://docs.rapids.ai/api/libcudf/stable/group__copy__concatenate.html#ga172353e7673decae3d2ebad09e003832

Reply:
So apparently the creation of a table with mismatch columns also does this check.

`NODE_CUDF_EXPECT(col->size() == num_rows_, "All Columns must be of same length");`
in `table.cpp` `Table::Table(...)`

So that ends up throwing before it even gets to my `concat` implementation.

Reply:
isn't the error condition if the table views have differing number columns (not about the length columns)?

Reply:
spoke with Bryan and he cleared up my misunderstandings...

1. Using the Table constructor and providing columns of mismatched sizes leads to CUDF blowing up. I'll handle this in the PR with a try catch
2. The public method `concat` inside of `Table.ts` can be called with an array of tables of potentially mismatched sizes. We should also wrap this in a try .. catch.

A side conversation emerged of whether or not we want to allow users to access `Table.ts` & `Column.ts`, I think we have an issue posted about this.